### PR TITLE
table: add disabled state to table

### LINKED ISSUE #44: Add disabled state to table head when the column is orderable

### PR DESCRIPTION
**Pull request by LeoNero**
Closes #44

Depends on https://github.com/pagarme/former-kit-skin-pagarme/pull/33

### CODE REVIEW

**Comment by derekstavis** [comment]
Hey @LeoNero, thanks for the changes. This was a needed one!!

I believe making the table heading disabled is not sufficient. I'd also block row click events and selection (checkboxes). Row expansion could be also disabled, as we can't guarantee that next results will match the expanded ones.

**Comment by LeoNero** [comment]
@Deivis I would change the commit's context to `table`. Also, I think we could implement @derekstavis 's suggestions :)

File: src/Table/TableExpandedRow.js [hunk 1]
```diff
@@ -154,6 +167,10 @@ TableExpandedRow.propTypes = {
    * Set of data native of row data from the table.
    */
   data: shape({}).isRequired,
+  /**
+   * Set of data native of row data from the table.
```
Comment (uselessdev) [inline src/Table/TableExpandedRow.js]: Wrong docs

**Comment by LeoNero** [comment]
@lucasjs I think this is the correct functionality, since a not orderable column is not clickable, there is no need to disable it

File: src/Table/TableExpandedRow.js [hunk 2]
```diff
@@ -104,18 +104,31 @@ class TableExpandedRow extends PureComponent {
...
     } = this.props
-
     const cols = columns.filter(col => !col.isAction).map(this.renderColumn)
     const actions = columns.filter(col => col.isAction).map(this.renderAction)
+    const rowClasses = classNames(
+      className,
+      theme[parity],
+      theme.expandedRow,
+      {
+        [theme.disabled]: disabled,
```
Comment (LeoNero) [inline src/Table/TableExpandedRow.js]: Add this class to the theme prop :)

File: src/Table/TableRow.js [hunk 3]
```diff
@@ -171,29 +172,35 @@ class TableRow extends Component {
       className,
       {
         [theme.clickable]: clickable,
+        [theme.disabled]: disabled,
```
Comment (LeoNero) [inline src/Table/TableRow.js]: Add this class to the theme prop :)

File: src/Table/TableRow.js [hunk 4]
```diff
@@ -258,6 +266,10 @@ TableRow.propTypes = {
    * Set of data native of row data from the table.
    */
   data: shape({}).isRequired,
+  /**
+   * indicates that the line is not interactive
```
Comment (LeoNero) [inline src/Table/TableRow.js]: `i` should be capitalized

File: src/Table/TableHead.js [hunk 5]
```diff
@@ -208,6 +218,10 @@ TableHead.propTypes = {
    * Selectable column checkbox state.
    */
   selected: bool,
+  /**
+   * Disable the click on orderable columns
```
Comment (LeoNero) [inline src/Table/TableHead.js]: `Disables`

File: src/Table/TableExpandedRow.js (hunk 1 above)
Comment (Deivis) [reply src/Table/TableExpandedRow.js]: done

File: src/Table/TableRow.js (hunk 3 above)
Comment (Deivis) [reply src/Table/TableRow.js]: done

File: src/Table/TableExpandedRow.js (hunk 2 above)
Comment (Deivis) [reply src/Table/TableExpandedRow.js]: done

**Pull request merged**